src/future/poll_fn.rs

```diff
@@ -23,15 +23,18 @@ use crate::task::{Context, Poll};
 /// #
 /// # })
 /// ```
-pub async fn poll_fn<F, T>(f: F) -> T
```

Comment:
Why not use `async` syntax ?

Reply:
the current async-await requires `std`, so cannot use in no-std environment